### ISSUE TITLE
Fix typing for Vue date picker component

### ISSUE THREAD

**Issue by shoxieje** [schedule-x/schedule-x#540]
**Description**
I would like to request the addition of a wrapper component or a dedicated Vue 3 component to render the DatePicker in the @schedule-x/date-picker library. This feature would greatly enhance the usability and integration of the DatePicker within Vue 3 applications.

**Justification**
The DatePicker is a highly useful component for many projects, but integrating it into Vue 3 applications requires additional effort to create custom wrappers. Providing an official wrapper component would streamline this process, making it easier for developers to integrate and use the DatePicker in their Vue 3/Nuxt 3 projects. This would also ensure better consistency and compatibility with the framework’s ecosystem.

**Comment by tomosterlund** [comment]
Just released this via `@schedule-x/vue`. Haven't added full documentation for the component yet, but you can find a minimal docu for it here: https://github.com/schedule-x/vue

**Comment by shoxieje** [comment]
Hi Tom, is it possible to add date-picker-options onto the SxDatePicker wrapper like the ScheduleXCalendar has :calendar-app? Something like <SxDatePicker :date-picker-app /> to import the options. Thanks a lot!

**Comment by tomosterlund** [comment]
Hi! You should be able to use all options from the date picker configuration, except `selectedDate` and `listeners.onChange`

![Bildschirmfoto 2024-07-16 um 06 09 52](https://github.com/user-attachments/assets/494a8863-fbfb-457b-a8f1-fdc9a0346618)

If you use TS, are you not getting the correct typing for this?

**Comment by shoxieje** [comment]
Yes I don't think typing work on SxDatePicker. However, model-value and config works though. Not sure why but when I read the import file there aren't no ./sx-date-picker on @schedule-x/vue/dist/types/index.d.ts (First Image). I've updated all the packages to their latest version. The second image shows that the imported SxDatePicker component is any. The SxDatePicker is imported from: import { SxDatePicker } from '@schedule-x/vue' 
<img width="475" alt="Screenshot 2024-07-16 at 9 39 18 PM" src="https://github.com/user-attachments/assets/e00c229d-1352-431c-921c-5eb4a6b8fe5a">
<img width="626" alt="Screenshot 2024-07-16 at 9 41 00 PM" src="https://github.com/user-attachments/assets/f6c0d8c5-d978-4907-b12a-fda70b798ec8">

**Comment by tomosterlund** [comment]
The type generation was broken due to a TS error which unfortunately passed through the build without exit code.

Fixed this in the latest version
